### PR TITLE
Enumse

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/misdat.h [hunk 1]
```diff
@@ -9,6 +9,12 @@
 
 namespace devilution {
 
+enum mienemy_type : uint8_t {
```
Comment (julealgon) [inline Source/misdat.h:12]: Why name it like this? Can't we use a more meaningful name like `MissileTarget`?

Comment (julealgon) [inline Source/misdat.h:12]: Can't we use `enum class` for new enums?

**Review by julealgon — CHANGES_REQUESTED** [review]

**Review by glebm — APPROVED** [review]

File: Source/misdat.h (hunk 1 above)
Comment (glebm) [reply Source/misdat.h:12]: Yeah we should

One big advantage of `enum class` other than scoping is that it's not implicitly convertible to int (requires a `static_cast`)

Comment (AJenbo) [reply Source/misdat.h:12]: It's the current name, I don't feel like renaming things atm as you know my spelling is piss poor so it both take longer and is significantly more likely to contain mistakes. Hopefully some one else will rename things in to sensible names using our chosen code style (camel case). I like your suggested name, feel free to open a PR with it.

Comment (AJenbo) [reply Source/misdat.h:12]: sure, for new ones 🤷

**Pull request merged**